packages/loot-core/src/server/budget/goaltemplates.js

```diff
@@ -333,6 +350,42 @@ async function applyCategoryTemplate(category, template_lines, month, force) {
       case 'error':
         return null;
       default:
+      case 'schedule': {
+        let { id: schedule_id } = await db.first(
+          'SELECT id FROM schedules WHERE name = ?',
+          [template.name],
+        );
+        let rule = await getRuleForSchedule(schedule_id);
+        let conditions = rule.serialize().conditions;
+        let { date: dateCond, amount: amountCond } =
+          extractScheduleConds(conditions);
+        let next_date_string = getNextDate(dateCond, current_month);
+        let num_months = differenceInCalendarMonths(
+          new Date(next_date_string),
+          current_month,
+        );
+        let target = -getScheduledAmount(amountCond.value);
+        let diff = target - last_month_balance;
```

Comment:
I think this should probably reflect "this months" balance.  There could potentially be spending present for the current month that wouldn't be taken into account.

Reply:
How would you feel about this?  It uses this months balance and the already budgeted amount to determine what's needed.
```suggestion
        let diff = target - balance + budgeted;
```

Reply:
Sounds good, this is probably more in line what a "layman" would expect the template to do.